Source/Initialization/DivCleaner/ProjectionDivCleaner.cpp

```diff
@@ -171,31 +188,59 @@ ProjectionDivCleaner::solve ()
 
     for (int ilev = 0; ilev < m_levels; ++ilev)
     {
-        MLPoisson mlpoisson({geom[ilev]}, {ba[ilev]}, {dmap[ilev]}, info);
-
-        mlpoisson.setMaxOrder(m_linop_maxorder);
-        mlpoisson.setDomainBC(lobc, hibc);
+        if (m_grid_type == GridType::Collocated) {
```

Comment:
These parts can be simplified a bit by setting `linop` to an uninitialized pointer for a linear solver. That way only the initialization of the linear solver has to be in the `if` block and the rest of the code does not have to be duplicated.

Reply:
If it helps, this can be used as an example of how the above is done elsewhere:
https://github.com/BLAST-WarpX/warpx/blob/480c53de3d95649a9774c76b764678b5ff735d61/Source/ablastr/fields/PoissonSolver.H#L309

Reply:
I tried this, but since the common inherited linear operator is the base MLLinOp class and the interface is different between cell centered and nodal operators, it would require reworking the class structure and virtual function structure of MLMG and the linear operator classes as a whole. I think having a few duplicated lines of code is OK here, even though it irks me a bit.

Reply:
OK, I ended up setting up a private templated member function to remove duplicated code.